### PR TITLE
Set correct values for allow_all, deny_all, and enforce

### PR DESCRIPTION
**Pull request by soerenmartius**
Values for `allow_all`, `deny_all` and `enforce` must be strings rather than booleans. User facing API should expose booleans though

### CODE REVIEW

File: README.tfdoc.hcl [hunk 1]
```diff
@@ -164,14 +164,14 @@ section {
           END
 
           attribute "allow_all" {
-            type        = bool # TODO: should it be string "TRUE"?
```
Comment (waxb) [inline README.tfdoc.hcl:167]: according to the resource schema these attributes are strings:
```
                        "allow_all": {
                                Type:        schema.TypeString,
                                Optional:    true,
                                Description: "Setting this to true means that all values are allowed. This field can be set only in Policies for list constraints.",
                        },

                        "deny_all": {
                                Type:        schema.TypeString,
                                Optional:    true,
                                Description: "Setting this to true means that all values are denied. This field can be set only in Policies for list constraints.",
                        },

                        "enforce": {
                                Type:        schema.TypeString,
                                Optional:    true,
                                Description: "If `true`, then the `Policy` is enforced. If `false`, then any configuration is acceptable. This field can be set only in Policies for boolean constraints.",
                        },
```

if we have the type as `bool` the module would be opinionated. for simplicity i'd go with `bool` anyway but then please see my comment below.

File: main.tf [hunk 2]
```diff
@@ -12,9 +12,9 @@ resource "google_org_policy_policy" "policy" {
         for_each = try(spec.value.rules, [])
 
         content {
-          allow_all = try(rules.value.allow_all, null)
-          deny_all  = try(rules.value.deny_all, null)
-          enforce   = try(rules.value.enforce, null)
+          allow_all = try(rules.value.allow_all, null) != null ? rules.value.allow_all == true ? "TRUE" : null : null
```
Comment (waxb) [inline main.tf:15]: if `*_all` and `enforce` are booleans, this could be:
```
allow_all = try(rules.value.allow_all, null) != null ? rules.value.allow_all ? "TRUE" : "FALSE" : null
```

File: main.tf [hunk 3]
```diff
@@ -12,9 +12,9 @@ resource "google_org_policy_policy" "policy" {
...
+          deny_all  = try(rules.value.deny_all, null) != null ? rules.value.deny_all == true ? "TRUE" : null : null
```
Comment (waxb) [inline main.tf:16]: if `*_all` and `enforce` are booleans, this could be:
```
deny_all = try(rules.value.deny_all, null) != null ? rules.value.deny_all ? "TRUE" : "FALSE" : null
```

File: main.tf [hunk 4]
```diff
@@ -12,9 +12,9 @@ resource "google_org_policy_policy" "policy" {
...
+          enforce   = try(rules.value.enforce, null) != null ? rules.value.enforce == true ? "TRUE" : null : null
```
Comment (waxb) [inline main.tf:17]: if `*_all` and `enforce` are booleans, this could be:
```
enforce = try(rules.value.enforce, null) != null ? rules.value.enforce ? "TRUE" : "FALSE" : null
```

File: main.tf (hunk 2 above)
Comment (soerenmartius) [reply main.tf:15]: hm this is what I tried (more or less) in https://github.com/mineiros-io/terraform-google-organization-policy/actions/runs/3401979187/jobs/5657434342 which caused the tests to fail

Comment (mariux) [inline main.tf:15]: why is this needed? can you show an error of the previous implementation? terraform should convert boolean to string just fine..

**Review by mariux — CHANGES_REQUESTED** [review]
